project/src/pages/login-page/login-page.tsx

```diff
@@ -1,28 +1,29 @@
+import { Navigate } from 'react-router-dom';
+import Header from '../../components/header/header';
+import { useAppDispatch, useAppSelector } from '../../hooks';
+import { AuthorizationStatus, AppPath } from '../../settings';
+import { login } from '../../store/action-creaters-middleware';
+
 export default function LoginPage(): JSX.Element {
-  return (
+  const dispatch = useAppDispatch();
+  const isAuthorized =
+    useAppSelector((store) => store.reducer.authorizationStatus) ===
+    AuthorizationStatus.Yes;
+  const onSubmitHandler = (evt: React.SyntheticEvent<HTMLFormElement>) => {
+    evt.preventDefault();
+    const formData = new FormData(evt.currentTarget);
+    const authData = {login: String(formData.get('email')), password: String(formData.get('password'))};
```

Comment:
```suggestion
    const {login, password}= validate(new FormData(evt.currentTarget));
    const authData = {login, password};
```

```ts
const goThrow   = (message: string)=>{throw new Error(message);};
const assertString = (value: any): string => typeof value === 'string'?value:goThrow('not a string');
const validate = (form:FromData)=>{
  return {
    login: assertString(form.get('email')),
    password: assertString(from.get('password')),
   };
}
```

идея в том, что если случайно имя пользователя заменят на input[type=file] "сразу" заметишь